test/end2end_test.go

```diff
@@ -7668,3 +7668,82 @@ func (s) TestClientSettingsFloodCloseConn(t *testing.T) {
 	s.GracefulStop()
 	timer.Stop()
 }
+
+// TestDeadlineSetOnConnectionOnClientCredentialHandshake tests that there is a deadline
+// set on the net.Conn when a credential handshake happens in http2_client.
+func (s) TestDeadlineSetOnConnectionOnClientCredentialHandshake(t *testing.T) {
+	lis, err := net.Listen("tcp", "localhost:50053")
```

Comment:
You cannot hard code a port here or the test will not pass in some environments.  Look at other `Listen` calls in our tests for examples.

Reply:
Changed to "localhost:0".